datajoint/logging.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import os
+import sys
+import io
+
+logger = logging.getLogger(__name__.split(".")[0])
+
+log_level = os.environ.get("DJ_LOG_LEVEL", "warning").upper()
+
+log_format = logging.Formatter(
+    "[%(asctime)s][%(funcName)s][%(levelname)s]: %(message)s"
+)
+
+stream_handler = logging.StreamHandler()  # default handler
+stream_handler.setFormatter(log_format)
+
+logger.setLevel(level=log_level)
+logger.handlers = [stream_handler]
+
+
+def excepthook(exc_type, exc_value, exc_traceback):
+    if issubclass(exc_type, KeyboardInterrupt):
+        sys.__excepthook__(exc_type, exc_value, exc_traceback)
+        return
+
+    if logger.getEffectiveLevel() == 10:
+        logger.debug(
+            "Uncaught exception", exc_info=(exc_type, exc_value, exc_traceback)
+        )
+    else:
+        logger.error(f"Uncaught exception: {exc_value}")
+
+
+sys.excepthook = excepthook
+
+
+# https://github.com/tqdm/tqdm/issues/313#issuecomment-267959111
+class TqdmToLogger(io.StringIO):
```

Comment:
Where is this used? Does TQDM need to output anything to logger?

Reply:
Agree that we actually don't need this. We don't need to manage `tqdm`'s logs.